randovania/games/samus_returns/generator/bootstrap.py

```diff
@@ -37,14 +38,9 @@ def is_dna_node(node: PickupNode, config: BaseConfiguration) -> bool:
             return True
     # Boss pickups/locations
     elif artifact_config.prefer_bosses:
-        if config.final_boss == FinalBossConfiguration.ARACHNUS:
-            _boss_indices.pop(0)
-        elif config.final_boss == FinalBossConfiguration.DIGGERNAUT:
-            _boss_indices.pop(2)
-        elif config.final_boss == FinalBossConfiguration.QUEEN:
-            _boss_indices.pop(3)
-        elif config.final_boss == FinalBossConfiguration.RIDLEY:
-            _boss_indices.pop(4)
+        for boss, index in _boss_mapping:
+            if config.final_boss.value == boss:
+                _boss_indices.pop(index)  # type: ignore
```

Comment:
why is this a type ignore

Reply:
Myoy was complaining

Reply:
then fix it, and dont just ignore it.

Reply:
I don't know what it wants 

```[mypy] reported by reviewdog 🐶
Argument 1 to "pop" of "list" has incompatible type "object"; expected "SupportsIndex"  [arg-type]
```

Reply:
```py
_boss_mapping = [{"Arachnus": 0}, {"Diggernaut": 2}, {"Queen": 3}, {"Ridley": 4}]
```

Reply:
and then change the for loop to be _boss_mapping.items()

Reply:
Ugh how'd I not see that

Reply:
I considered having the boss list only have the chase index, then appending all the valid indices depending on the final boss.